buildenv/jenkins/upstream_openjdk_tests

```diff
@@ -0,0 +1,306 @@
+    parameters {
+        string(name:'VERSIONS', defaultValue: '8 11', description: "Space separated list of versions to check")
+        string(name:'GH_USERNAME', description: "Github API username to authenticate as")
+        string(name:'GH_TOKEN', description: "Github API OAuth personalized access token to be used for authentication")
+        booleanParam(name:'INCLUDE_EA', defaultValue: true, description: "Whether or not to include EA builds")
+        booleanParam(name:'FORCE_RETEST', defaultValue: false, description: "Whether or not to force a re-test irrespective previous runs")
+    }
+
+    // All params have sensible defaults
+    List versions = params.VERSIONS ? params.VERSIONS.split(" ") : "8 11".split(" ")
+    boolean includeEA = params.INCLUDE_EA ? params.INCLUDE_EA : true
+    boolean forceRetest = params.FORCE_RETEST ? params.FORCE_RETEST : false
+    String user = params.GH_USERNAME ? params.GH_USERNAME : "gh-user-not-set"
+    String token = params.GH_TOKEN ? params.GH_TOKEN : "gh-token-not-set"
+    String userTokenBase64 = Base64.getEncoder().encodeToString((user + ":" + token).getBytes())
+    Map releaseCtxts = [:]
+    Map latestRelInfos = [:]
+
+    TEST_JOB_NAME = "${env.JOB_NAME}"
+    File workspace = new File("${env.WORKSPACE}")
+    workspace.mkdirs()
+    versions.each { version ->
+        ReleaseRetriever retriever = new ReleaseRetriever(version, includeEA, userTokenBase64)
+        List ctxts = retriever.getReleaseContexts()
+        latestRelInfos.put(version, new LatestReleaseInfo(workspace, version, retriever.getLatestReleaseDate()))
+        releaseCtxts.put(version, ctxts)
+    }
+    dir("$env.WORKSPACE") { // use workspace relative directory
+        versions.each { version ->
+            stage("prepare OpenJDK Version: $version") {
+                LatestReleaseInfo relInfo = latestRelInfos.get(version)
+                List ctxts = releaseCtxts.get(version)
+                ctxts.each { println it.toString() }
+                if (forceRetest || relInfo.isRetestNeeded()) {
+                    // trigger tests for each context
+                    ctxts.each{ r -> 
+                        stage("prepare $r.OS-$r.ARCH") {
+                            def jdk_url = r.JDK_URL
+                            def source_url = r.SOURCE_URL
+                            def jdk_version = version
+                            def jenkins_file = r.getJenkinsFile()
+                            def jdk_jre_url = "$r.JDK_URL $r.JRE_URL"
+                            parallelJobs["Test_openjdk${version}_hs_sanity.openjdk_${r.ARCH}_${r.OS}"] = {
+                                // Run a _sanity.openjdk test job with appropriate parameters
+                                build job: TEST_JOB_NAME, parameters: [
+                                    string(name: 'ADOPTOPENJDK_REPO', value: params.ADOPTOPENJDK_REPO ? params.ADOPTOPENJDK_REPO : "https://github.com/AdoptOpenJDK/openjdk-tests.git"),
```

Comment:
I wondered if this is specific for upstream releases to run sanity.openjdk or sanity.system tests we should not need all those extra parameters to make it more clear. Those extra parameters are specific for grinder, i.e. for developers' personal build.  Parameters like BUILD_LIST, TARGET, JDK_IMPL, SDK_RESOURCE, CUSTOMIZED_SDK_URL &CUSTOMIZED_SDK_SOURCE_URL should be enough.

Reply:
Sure. I've kept them in as I wanted a way to test changes prior merge.

`JDK_VERSION` is needed too. As otherwise, it would inherit the default value from `Grinder` or `Grinder_Sandbox` which is `8` causing a conflict with `AUTO_DETECT=true` on JDK 11.

Reply:
I see. The configuration of Grinder and Grinder_Sandbox is slightly different. No default JDK_VERSION value in Grinder and default value 8 in Grinder_sandbox. As AUTO_DETECT is default true JDK_VERSION should not set default value. I just removed default JDK_VERSION in Grinder_Sandbox.

Reply:
Same as JDK_IMPL, I've removed the default value.

Reply:
@jerboaa if you see the job configuration https://ci.adoptopenjdk.net/view/Test_upstream/job/Test_upstream_openjdk8_hs_sanity.openjdk_x86-64_linux/configure, all those parameter has been set default value as you needed! Thanks @smlambert